kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/k8s/cmd/BaseCmdKubeClient.java

```
@@ -0,0 +1,146 @@
+import io.kroxylicious.systemtests.executor.ExecResult;
+
+import static java.util.Arrays.asList;
+
+/**
+ * The type Base cmd kube client.
+ *
+ * @param <K>  the type parameter
+ */
+public abstract class BaseCmdKubeClient<K extends BaseCmdKubeClient<K>> implements KubeCmdClient<K> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseCmdKubeClient.class);
+
+    private static final String APPLY = "apply";
+    private static final String DELETE = "delete";
+
+    /**
+     * The Namespace.
+     */
+    String namespace = defaultNamespace();
+
+    /**
+     * The type Context.
+     */
+    protected static class Context implements AutoCloseable {
+        @Override
+        public void close() {
+            // Do nothing
+        }
+    }
+
+    private static final Context NOOP = new Context();
+
+    /**
+     * Default context context.
+     *
+     * @return the context
+     */
+    protected Context defaultContext() {
+        return NOOP;
+    }
+
+    /**
+     * Namespaced command list.
+     *
+     * @param rest the rest
+     * @return the list
+     */
+    protected List<String> namespacedCommand(String... rest) {
+        return namespacedCommand(asList(rest));
+    }
+
+    private List<String> namespacedCommand(List<String> rest) {
+        List<String> result = new ArrayList<>();
+        result.add(cmd());
+        result.add("--namespace");
+        result.add(namespace);
+        result.addAll(rest);
+        return result;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public K apply(File... files) {
+        try (Context context = defaultContext()) {
+            Map<File, ExecResult> execResults = execRecursive(APPLY, files, Comparator.comparing(File::getName).reversed());
+            for (Map.Entry<File, ExecResult> entry : execResults.entrySet()) {
+                if (!entry.getValue().isSuccess()) {
+                    LOGGER.warn("Failed to apply {}!", entry.getKey().getAbsolutePath());
```

It is really safe to continue if this fails? Fail early is a good mantra.